Update Fan Cases with mention of the Pi 4 Case Fan (#3929)
raspberrypi/documentation: documentation/asciidoc/computers/raspberry-pi/frequency-management.adoc

=== Fan cases

> For the Raspberry Pi 4, you have the option of adding the https://www.raspberrypi.com/products/raspberry-pi-4-case-fan/[Raspberry Pi 4 Case Fan] to the lid of the Raspberry Pi 4 case.

Reviewer: Can we get an introduction sentence, instead of jumping straight into Raspberry Pi 4 info? Maybe something like: "To ensure the best performance for your Raspberry Pi, use an active cooling solution such as a fan."

Then, we could either break this up into "Raspberry Pi 4" and "Raspberry Pi 5" sections (slightly preferred), or perhaps provide info about each model in an unordered list. Either way, we should probably restructure this sentence introduction to better match the Raspberry Pi 5 equivalent:

```suggestion
For Raspberry Pi 4, add the https://www.raspberrypi.com/products/raspberry-pi-4-case-fan/[Raspberry Pi 4 Case Fan] to the lid of the Raspberry Pi 4 case.
```

And I'd probably modify the line below to something like "For Raspberry Pi 5, use one of the official fan options:"

Reviewer: Building on the hysteresis comment below, perhaps add a sentence like this to the introduction:

"Raspberry Pi firmware manages fan speeds for all official fans."

That way, it's clear that the Raspberry Pi 4 fan doesn't just blow at 100% all of the time :-)

Raspberry Pi 5 has two official fan options to assist with cooling:

* https://www.raspberrypi.com/products/active-cooler/[Active Cooler]
* https://www.raspberrypi.com/products/raspberry-pi-5-case/[Case for Raspberry Pi 5]

Both of the Raspberry Pi 5 fan options plug into the four-pin JST-SH PWM fan connector located in the upper right of the board between the 40-pin GPIO header and the USB 2 ports. The fan connector pulls from the same current limit as USB peripherals. We recommend the Active Cooler case for overclockers, since it provides better cooling performance.

> The fans of the Active Cooler and the Raspberry Pi 5 Case are both actively managed by Raspberry Pi firmware. As the temperature of the Raspberry Pi 5 increases, the fan reacts in the following way:

Reviewer: I believe the Raspberry Pi 4 Case Fan firmware also actively manages fan speed. Perhaps we could trim the first sentence here, and instead mention that idea in the introduction?
```suggestion
As the temperature of the Raspberry Pi 5 increases, official fans react in the following way:
```

At boot the fan is turned on, and the tachometer input is checked to see if the fan is spinning. If it is, then the `cooling_fan` device tree overlay is enabled. This overlay is in `bcm2712-rpi-5-b.dtb` by default, but with `status=disabled`.

> ==== Raspberry Pi 5 fan connector pinout

Author: One more nit 😁 Thoughts on this header? It's at the same level as the other "fan" headers (I didn't want to make it too small). Normally I don't think I would separate this information out using a header, but in this case, the connector info is linked in the introduction.

Reviewer: Seems like a reasonable compromise. Ideally we'd nest this within the Pi 5 section, but... that might go too deep. This seems to be the best place for this right now.
